src/stdlib_stats_distribution_PRNG.fypp

```diff
@@ -0,0 +1,211 @@
+    function dist_rand_${t1[0]}$${k1}$(n) result(res)
+    !! Random integer generation for various kinds
+    !! result = [-2^k, 2^k - 1], k = 7, 15, 31, 63, depending on input kind
+    !! Result is used as bit model number instead of regular arithmetic number
+    !!
+        ${t1}$, intent(in) :: n
+        ${t1}$ :: res
+        integer :: k
+
+        k = MAX_INT_BIT_SIZE - bit_size(n)
+        res = shiftr(xoshiro256ss( ), k)
+    end function dist_rand_${t1[0]}$${k1}$
+
+    #:endfor
+
+    function xoshiro256ss( ) result (res)
+    ! Generate random 64-bit integers
+    ! Written in 2018 by David Blackman and Sebastiano Vigna (vigna@acm.org)
+    !
+    ! This is xoshiro256** 1.0, one of our all-purpose, rock-solid
+    ! generators. It has excellent (sub-ns) speed, a state (256 bits) that is
+    ! large enough for any parallel application, and it passes all tests we
+    ! are aware of.
+    !
+    ! The state must be seeded so that it is not everywhere zero. If you have
+    ! a 64-bit seed, we suggest to seed a splitmix64 generator and use its
+    ! output to fill st.
+    !
+    ! Fortran 90 version translated from C by Jim-215-Fisher
+    !
+        integer(int64) :: res, t
+
+        if(.not. seed_initialized) call random_distribution_seed_iint64(si,t)
+        res = rol64(st(2) * 5 , 7) * 9
+        t = shiftl(st(2), 17)
+        st(3) = ieor(st(3), st(1))
+        st(4) = ieor(st(4), st(2))
+        st(2) = ieor(st(2), st(3))
+        st(1) = ieor(st(1), st(4))
+        st(3) = ieor(st(3), t)
+        st(4) = rol64(st(4), 45)
+        return
+    end function xoshiro256ss
+
+    function rol64(x, k) result(res)
+        integer(int64), intent(in) :: x
+        integer, intent(in) :: k
+        integer(int64) :: t1, t2, res
+
+        t1 = shiftr(x, (64 - k))
+        t2 = shiftl(x, k)
+        res = ior(t1, t2)
+        return
+    end function rol64
+
+
+    subroutine jump
+    ! This is the jump function for the xoshiro256ss generator. It is equivalent
+    ! to 2^128 calls to xoshiro256ss(); it can be used to generate 2^128
+    ! non-overlapping subsequences for parallel computations.
+    ! Written in 2018 by David Blackman and Sebastiano Vigna (vigna@acm.org)
+    !
+    ! Fortran 90 version translated from C by Jim-215-Fisher
+    integer(int64) :: jp(4) = [1733541517147835066_int64,                       &
+                               -3051731464161248980_int64,                      &
+                               -6244198995065845334_int64,                      &
+                               4155657270789760540_int64]
+    integer(int64) :: s1 = 0, s2 = 0, s3 = 0, s4 = 0, c = 1_int64
+    integer :: i, j, k
+
+    do i = 1, 4
+        do j = 1, 64
+            if(iand(jp(i), shiftl(c, j - 1)) /= 0) then
+                s1 = ieor(s1, st(1))
+                s2 = ieor(s2, st(2))
+                s3 = ieor(s3, st(3))
+                s4 = ieor(s4, st(4))
+            end if
+            k = xoshiro256ss( )
```

Comment:
Where is `k` used in this subroutine?

Reply:
k is never used. But calling xoshiro256ss function will change the saved state randomly.